### ISSUE TITLE
Yolov5s model conversion and segmentation fault 

### ISSUE THREAD

**Issue by emreaniloguz** [Tencent/ncnn#4195]
## error log | 日志或报错信息 | ログ

Segmentation fault (core dumped)

## model | 模型 | モデル
1. original model
Yolov5s

## how to reproduce | 复现步骤 | 再現方法
1. Download my .bin and .param file 
2. Run the code that I wrote (it differs original ncnn documentation slightly)

I have trained a custom Yolo5s model (from here: https://github.com/ultralytics/yolov5),

Then I converted .pt file to .onnx format with the code that i wrote 

```
import torch
import torch.onnx

model = torch.hub.load('ultralytics/yolov5', 'custom', path="best.pt",force_reload=True)  # default
model = model.cpu()
dummy_input = torch.randn(1, 3, 416, 416)
torch.onnx.export(model, dummy_input, "yolov5s_mobile_phone.onnx")
```

Then I simplified the onnx model, 

```
onnxsim ../yolov5s_mobile_phone.onnx ../yolov5s_mobile_phone_simplified.onnx
```
Finally, I converted the simplified onnx model to the .bin and .param format. 
```
onnx2ncnn yolov5s_mobile_phone_simplified.onnx yolov5s.bin yolov5s.param
``` 

With the operations above I follow this guide to get an output from my model but, 
```
#include<iostream>
#include <opencv2/opencv.hpp>

#include"net.h"


int main(){
    std::cout << "Hello World!" << std::endl;
    
    
    cv::Mat image = cv::imread("mfagan.png");

    if (image.empty()){
        std::cout << "Image not found" << std::endl;
        return 1;
    }

    int w = image.cols;
    int h = image.rows;

    //resize image to network input size


    ncnn::Mat in = ncnn::Mat::from_pixels_resize(image.data, ncnn::Mat::PIXEL_RGB, w, h, 416, 416);    //float mean[1] = { 128.f };
    float mean[1] = { 128.f };
    float norm[1] = { 1/128.f };
    in.substract_mean_normalize(mean, norm);

    ncnn::Net net;

    net.load_param("yolov5s.param");
    net.load_model("yolov5s.bin");

    ncnn::Extractor ex = net.create_extractor();
    ex.set_light_mode(true);
    ex.set_num_threads(4);



    ex.input("onnx::Cast_0", in);
    ncnn::Mat feat;
    ex.extract("676", feat);


    return 0;


}

```

The error

```
Hello World!
Segmentation fault (core dumped)
```

**Comment by emreaniloguz** [comment]
You can download files from here;

https://drive.google.com/drive/folders/1BvyEBIf1n-8i7NRU5qznxPUQdLtbW9dH?usp=sharing

**Comment by xudahong** [comment]
hello，have you solved this problem? i had the same problem.

**Comment by emreaniloguz** [comment]
> hello，have you solved this problem? i had the same problem.

Yeah, I have solved the problem. I'm trying to prepare a repo about this issue, when it is finished I will be sharing it from here.

**Comment by xudahong** [comment]
> > hello，have you solved this problem? i had the same problem.
> 
> Yeah, I have solved the problem. I'm trying to prepare a repo about this issue, when it is finished I will be sharing it from here.

ok,thanks a lot

**Comment by dtiny** [comment]
> > hello，have you solved this problem? i had the same problem.
> 
> Yeah, I have solved the problem. I'm trying to prepare a repo about this issue, when it is finished I will be sharing it from here.

hello, i had the same problem. could you share to solve it.

**Comment by pastukhov-aleksandr** [comment]
> > hello，have you solved this problem? i had the same problem.
> 
> Yeah, I have solved the problem. I'm trying to prepare a repo about this issue, when it is finished I will be sharing it from here.

could you share to solve it. Pls!

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
